src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -364,15 +364,17 @@ static ucs_status_t uct_mlx5_iface_init_dm(uct_rc_mlx5_iface_common_t *iface,
                                               uct_mlx5_dm_data_t,
                                               uct_mlx5_iface_dm_device_cmp,
                                               uct_mlx5_iface_init_dm_tl, rc_iface, config);
-        if (iface->dm.dm && !iface->dm.dm->dm) {
+        if (iface->dm.dm == NULL) {
```

Comment:
minor: you may avoid extra nesting, like:
`if (config->dm.seg_len * config->dm.count == 0) return/goto`

Reply:
updated